src/common/protocols/master_metadata_service_client.cc

```diff
@@ -0,0 +1,59 @@
+#include "src/common/protocols/master_metadata_service_client.h"
```

Comment:
Is there a reason why we are still keeping this here, in "protocols"? I find it a little bit confusing. As mentioned in previous comment will be nice to have the service client imp in same dir as service impl. Makes it easier to find. Maybe somewhere else, but "protocols" confuses me here. It's not intuitive for me to check the protocols dir when I am looking for the service client impl code. Atleast for me. But don't want to block this PR. Will approve.